cpp/refactor_module/algorithm/refactor.cpp

```diff
@@ -1 +1,55 @@
 #include "refactor.hpp"
+
+#include <mg_utils.hpp>
+#include <unordered_set>
+
+void Refactor::InsertCloneNodesRecord(mgp_graph *graph, mgp_result *result, mgp_memory *memory, const int cycle_id,
+                                      const int node_id) {
+  auto *record = mgp::result_new_record(result);
+
+  mg_utility::InsertIntValueResult(record, std::string(kResultClonedNodeId).c_str(), cycle_id, memory);
+  mg_utility::InsertNodeValueResult(graph, record, std::string(kResultNewNode).c_str(), node_id, memory);
+}
+
+void Refactor::CloneNodes(mgp_list *args, mgp_graph *memgraph_graph, mgp_result *result, mgp_memory *memory) {
+  mgp::memory = memory;
+  const auto arguments = mgp::List(args);
+  auto graph = mgp::Graph(memgraph_graph);
+  try {
+    const auto nodes = arguments[0].ValueList();
+    const auto clone_rels = arguments[1].ValueBool();
+    const auto skip_props = arguments[2].ValueList();
+    std::unordered_set<std::string_view> skip_props_searchable;
+    for (auto element : skip_props) {
+      skip_props_searchable.insert(element.ValueString());
+    }
```

Comment:
Can you open PR for this on Memgraph: https://memgraph.slack.com/archives/CSR01RS80/p1691057006080239?thread_ts=1691053492.934529&cid=CSR01RS80

We should add this traits under iterator and it should work in my opinion.